appsec/src/helper/client.cpp

```diff
@@ -415,91 +372,70 @@ bool client::handle_command(network::config_sync::request & /* command */)
     return false;
 }
 
+template <typename T, bool actions>
+bool client::send_message(const std::shared_ptr<typename T::response> &message)
+{
+    if (!message) {
+        return false;
+    }
+
+    if (spdlog::should_log(spdlog::level::debug)) {
+        // NOLINTNEXTLINE(misc-const-correctness)
+        std::ostringstream all_verdicts;
+        if constexpr (actions) {
+            for (const auto &action : message->actions) {
+                all_verdicts << action.verdict << " ";
+            }
+            if (message->actions.empty()) {
+                all_verdicts << "no verdicts";
+            }
+        }
+        SPDLOG_DEBUG("sending response to {}, verdicts: {}",
+            message->get_type(), all_verdicts.str());
```

Comment:
This log doesn't work properly when we respond with a `config_features`, because we're sending a response to `request_init` but it says sending response to `config_features`. You could make the `if constexpr (actions)` also produce a different log message since there are no verdicts.

Reply:
This is just a debug message. When this scenario happen, there is a message before hitting this function saying `config_features` messaged is replying to a `request_init`. Otherwise I would need to add a new param just for debugging purposes. Let me know if you think that's still required

Reply:
That's fine, but the other issue can be fixed such as:

```cpp
        if constexpr (actions) {
            std::ostringstream all_verdicts;
            for (const auto &action : message->actions) {
                all_verdicts << action.verdict << " ";
            }
            if (message->actions.empty()) {
                all_verdicts << "no verdicts";
            }
            SPDLOG_DEBUG("sending response to {}, verdicts: {}",
                message->get_type(), all_verdicts.str());
        } else {
            SPDLOG_DEBUG("sending response to {}", message->get_type());
        }
```